### PR TITLE
Feature: Added archive, patch and get project endpoints

### PR DESCRIPTION
**Pull request by BrandonDR**
Updated project type definitions
Switched from bearer to x-api-key header
Fixed PATCH requests

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -55,14 +55,14 @@ export default class DatanestClient {
...
+            'X-API-Key': this.apiKey,
         };
         if (this.clientId) {
             headers['X-Client-ID'] = this.clientId;
         }
         const options: DatanestRequestInit = {
             redirect: 'error',
-            mode: 'no-cors',
+            mode: method !== 'DELETE' && method !== 'PATCH' ? 'no-cors' : undefined,
             ...fetchOptions,
```
Comment (brijesh-datanest) [inline src/index.ts:66]: mode can be overridden by fetchOptions.

File: src/projects.ts [hunk 2]
```diff
@@ -158,4 +171,28 @@ export async function createProject(client: DatanestClient, projectData: Project
...
+    };
+}
+
+/**
+ * Archive a project to hide it from users, it will be automatically deleted after some time.
+ * @param client 
+ * @param projectUuid 
+ * @returns 
+ */
+export async function archiveProject(client: DatanestClient, projectUuid: string) {
```
Comment (brijesh-datanest) [inline src/projects.ts:195]: Why is delete being equated to archive ?

Comment (brijesh-datanest) [reply src/projects.ts:195]: nvm, understood this now.

Comment (BrandonDR) [reply src/projects.ts:195]: Allows us to restore if projects are accidentally deleted. We do not currently allow users to delete projects, they are instead deleted after being archived for 6 months, by default.

File: src/index.ts (hunk 1 above)
Comment (BrandonDR) [reply src/index.ts:66]: Wanted to support all the `fetch` api options to end developers, they can override this library

**Pull request merged**